tools/engine_tool/lib/src/commands/test_command.dart

```diff
@@ -0,0 +1,85 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:engine_build_configs/engine_build_configs.dart';
+import 'package:path/path.dart' as p;
+
+import '../build_utils.dart';
+import '../gn_utils.dart';
+import '../proc_utils.dart';
+import '../worker_pool.dart';
+import 'command.dart';
+import 'flags.dart';
+
+/// The root 'test' command.
+final class TestCommand extends CommandBase {
+  /// Constructs the 'build' command.
+  TestCommand({
+    required super.environment,
+    required Map<String, BuilderConfig> configs,
+  }) {
+    builds = runnableBuilds(environment, configs);
+    debugCheckBuilds(builds);
+    argParser.addOption(
+      configFlag,
+      abbr: 'c',
+      defaultsTo: 'host_debug',
+      help: 'Specify the build config to use',
+      allowed: <String>[
+        for (final Build config in runnableBuilds(environment, configs))
+          config.name,
+      ],
+      allowedHelp: <String, String>{
+        for (final Build config in runnableBuilds(environment, configs))
+          config.name: config.gn.join(' '),
+      },
+    );
+  }
+
+  /// List of compatible builds.
+  late final List<Build> builds;
+
+  @override
+  String get name => 'test';
+
+  @override
+  String get description => 'Runs a test target';
+
+  @override
+  Future<int> run() async {
+    final String configName = argResults![configFlag] as String;
+    final Build? build =
+        builds.where((Build build) => build.name == configName).firstOrNull;
+    if (build == null) {
+      environment.logger.error('Could not find config $configName');
+      return 1;
+    }
+
+    final Map<String, TestTarget> allTargets = await findTestTargets(
+        environment,
+        Directory(p.join(environment.engine.outDir.path, build.ninja.config)));
+    final Set<TestTarget> selectedTargets =
+        selectTargets(argResults!.rest, allTargets);
+    final List<String> buildTargets = selectedTargets
+        .map<String>((TestTarget target) => target.label.substring(2))
+        .toList();
```

Comment:
Can we exit early if no targets are found? For example:

```
PS C:\Code\f\engine\src\flutter> et test //flutter/shell/platform/windows
[2024-03-22T09:39:41.331][host_debug_unopt: GN]: OK
[host_debug: ninja] 0.0% (0/1) Regenerating ninja files
[2024-03-22T09:39:45.924][host_debug: ninja]: OK
````

Notice `//flutter/shell/platform/windows` is an invalid selector (it should have been `//flutter/shell/platform/windows/...`), but one could misinterpret this output as tests passing.

Perhaps instead we could list all potential test targets:

```
PS C:\Code\f\engine\src\flutter> et test //flutter/shell/platform/windows
[2024-03-22T09:39:41.331][host_debug_unopt: GN]: OK

Could not find a test target that matches '//flutter/shell/platform/windows'!

Test targets:

  //flutter/display_list:display_list_builder_benchmarks
  //flutter/display_list:display_list_region_benchmarks
  //flutter/display_list:display_list_rendertests

````

Reply:
Done.

Instead of automatically listing all targets I include a message saying to run 'et query tests' to see a list.